### PR TITLE
Update servers article to rc1

### PR DESCRIPTION
**Pull request by shirhatti**

### CODE REVIEW

**Comment by danroth27** [comment]
@Tratcher @Erikre Please review

File: aspnet/fundamentals/servers.rst [hunk 1]
```diff
@@ -3,7 +3,7 @@ Servers
 
 By `Steve Smith`_
 
-ASP.NET 5 is completely decoupled from the web server environment that hosts the application. As released, ASP.NET 5 supports IIS and IIS Express, WebListener, and Kestrel web servers, which run on a variety of platforms. Developers and third party software vendors can create their own custom servers as well within which to host their ASP.NET 5 applications.
+ASP.NET 5 is completely decoupled from the web server environment that hosts the application. As released, ASP.NET 5 supports WebListener, and Kestrel web servers, which run on a variety of platforms. Developers and third party software vendors can create their own custom servers as well within which to host their ASP.NET 5 applications.
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: IIS at least deserves a mention here.

File: aspnet/fundamentals/servers.rst [hunk 2]
```diff
@@ -19,17 +19,16 @@ In this article:
...
 
-The default web host for ASP.NET applications developed using Visual Studio 2015 is IIS / IIS Express. The "Microsoft.AspNet.Server.IIS" dependency is included in ``project.json`` by default, even with the Empty web site template. Visual Studio provides support for multiple profiles, associated with IIS Express and any other ``commands`` defined in ``project.json``. You can manage these profiles and their settings in the Debug tab of your web application project's Properties menu.
+The default web host for ASP.NET applications developed using Visual Studio 2015 is IIS / IIS Express as a reverse proxy server for Kestrel. The "Microsoft.AspNet.Server.Kestrel" and "Microsoft.AspNet.IISPlatformHandler" dependencies are included in ``project.json`` by default, even with the Empty web site template. Visual Studio provides support for multiple profiles, associated with IIS Express and any other ``commands`` defined in ``project.json``. You can manage these profiles and their settings in the Debug tab of your web application project's Properties menu or from the ``launchSettings.json`` file.
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: Discussion of IIS and reverse proxying needs to come much earlier.  Also, do we ever mention HttpPlatformHandler?

File: aspnet/fundamentals/servers.rst [hunk 3]
```diff
@@ -39,16 +38,16 @@ I've configured the sample project for this article to support each of the diffe
...
 .. literalinclude:: servers/sample/ServersDemo/src/ServersDemo/project.json
 	:lines: 1-17
-	:emphasize-lines: 12-14
+	:emphasize-lines: 14-15
 	:linenos:
 	:language: javascript
 	:caption: project.json (truncated)
 
-The ``run`` command will execute the application via its ``void main()`` method defined in ``program.cs``. In this case, this has been set up to configure and start an instance of ``WebListener``. This is not a typical means of launching a server, but is shown to demonstrate the possibility (the `Music Store sample application <https://github.com/aspnet/MusicStore>`_ also demonstrates this option).
+The ``run`` command will execute the application via its ``void main()`` method defined in ``program.cs``. In this case, this has been set up to configure and start an instance of ``Kestrel``. This is not a typical means of launching a server, but is shown to demonstrate the possibility (the `Music Store sample application <https://github.com/aspnet/MusicStore>`_ also demonstrates this option).
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: I think we can remove any discussion of the commands, and launching a web app via program main is the new standard anyways.

File: aspnet/fundamentals/servers.rst [hunk 4]
```diff
@@ -62,59 +61,46 @@ ASP.NET defines a number of :doc:`request-features` which may be supported on di
...
 	  - Yes
 	  - Yes
-	  - No
 	* - ITlsTokenBindingFeature
 	  - Yes
-	  - Yes
 	  - No
 	  
-To add support for web sockets in Kestrel, use the `WebSocketMiddleware <https://github.com/aspnet/WebSockets/blob/c86b157ad3cd00e8848c4895fe29de2f9d81a0b4/src/Microsoft.AspNet.WebSockets.Server/WebSocketMiddleware.cs>`_
+To add support for web sockets, use the `WebSocketMiddleware <https://github.com/aspnet/WebSockets/blob/c86b157ad3cd00e8848c4895fe29de2f9d81a0b4/src/Microsoft.AspNet.WebSockets.Server/WebSocketMiddleware.cs>`_
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: Link to a milestone label rather than a commit

File: aspnet/fundamentals/servers.rst [hunk 5]
```diff
@@ -125,42 +111,52 @@ The ``project.json`` file shown above demonstrates how to pass the ``server.urls
...
-Then, ``hosting.ini`` can include the settings the server will use (including the server parameter, as well):
+Then, ``hosting.json`` can include the settings the server will use (including the server parameter, as well):
 
-.. code-block:: text
+.. code-block:: json
 
-	server=Kestrel
-	server.urls=http://localhost:5004
+	{
+	  "server": "Microsoft.AspNet.Server.WebListener",
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: Kestrel

File: aspnet/fundamentals/servers.rst [hunk 6]
```diff
@@ -125,42 +111,52 @@ The ``project.json`` file shown above demonstrates how to pass the ``server.urls
...
+In addition to specifying configuration options, the server hosting the application can be referenced programmatically via the `IApplicationBuilder interface <https://github.com/aspnet/HttpAbstractions/blob/e7bf0e71bb2ee6e08dca82d8d7485fc89e98d0b7/src/Microsoft.AspNet.Http.Abstractions/IApplicationBuilder.cs>`_, available in the ``Configure`` method in ``Startup``. ``IApplicationBuilder`` exposes Server Features of type `IFeatureCollection <https://github.com/aspnet/HttpAbstractions/blob/e7bf0e71bb2ee6e08dca82d8d7485fc89e98d0b7/src/Microsoft.AspNet.Http.Features/IFeatureCollection.cs>`_. ``IServerAddressesFeature`` only expose a ``Addresses`` property, but different server implementations may expose additional functionality. For instance, WebListener exposes ``AuthenticationManager`` that can be used to configure the server's authentication:
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: link to labels

Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: This makes it sound like WebListener exposes AuthenticationManager off of IServerAddressesFeature.

File: aspnet/fundamentals/servers.rst [hunk 7]
```diff
@@ -125,42 +111,52 @@ The ``project.json`` file shown above demonstrates how to pass the ``server.urls
...
+
+In ASP.NET 5, the web application is hosted by an external process outside of IIS. The HTTP Platform Handler is an IIS 7.5+ module which is responsible for process management of http listeners and to proxy requests to processes that it manages. 
+
+
 
 WebListener
 -----------
 
-WebListener is a Windows-only server that allows ASP.NET applications to be hosted outside of IIS. It runs directly on the `Http.Sys kernel driver <http://www.iis.net/learn/get-started/introduction-to-iis/introduction-to-iis-architecture>`_, and has very little overhead. It supports the same feature interfaces as IIS; in fact, you can think of WebListener as a library version of IIS.
+WebListener is a Windows-only application server for ASP.NET 5. It runs directly on the `Http.Sys kernel driver <http://www.iis.net/learn/get-started/introduction-to-iis/introduction-to-iis-architecture>`_, and has very little overhead. It supports the same feature interfaces as IIS; in fact, you can think of WebListener as a library version of IIS.
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: application server -> http server

Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: remove the comment about feature interfaces, as those are no longer relevant to IIS.

File: aspnet/fundamentals/servers.rst [hunk 8]
```diff
@@ -173,88 +169,63 @@ You can add support for WebListener to your ASP.NET application by adding the "M
 Kestrel
 -------
 
-Kestrel is a cross-platform web server based on `libuv <https://github.com/libuv/libuv>`_, a cross-platform asynchronous I/O library. Kestrel is open-source, and you can `view the Kestrel source on GitHub <https://github.com/aspnet/KestrelHttpServer>`_. Kestrel is a great option to at least include support for in your ASP.NET 5 projects so that your project can be easily run by developers on any of the supported platforms. You add support for Kestrel by including "Kestrel" in your project's dependencies listed in ``project.json``.
+Kestrel is a cross-platform web server based on `libuv <https://github.com/libuv/libuv>`_, a cross-platform asynchronous I/O library. Kestrel is open-source, and you can `view the Kestrel source on GitHub <https://github.com/aspnet/KestrelHttpServer>`_. You add support for Kestrel by including "Kestrel" in your project's dependencies listed in ``project.json``.
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: Microsoft.AspNet.Server.Kestrel

File: aspnet/fundamentals/servers.rst [hunk 9]
```diff
@@ -173,88 +169,63 @@ You can add support for WebListener to your ASP.NET application by adding the "M
...
 
 Learn more about working with Kestrel to create :doc:`/tutorials/your-first-mac-aspnet`.
 
 Choosing a server
 -----------------
 
-If you intend to deploy your application on a Windows server, you should prefer the Windows optimized servers (IIS and WebListener). IIS remains the best choice for typical web application deployments on Windows servers, and provides the richest set of features. Choose WebListener instead of IIS only if you need to host your application within your own process or service. Use Kestrel if you intend to develop on or deploy to non-Windows environments.
-
+If you intend to deploy your application on a Windows server, you should run IIS as a reverse proxy server that manages and proxies requests to Kestrel. If deploying on Linux, you should run a comparable reverse proxy server such as Apache or Nginx to proxy requests to Kestrel. Choose WebListener instead of Kestrel if you are deploying to a Windows environment and require Domain Authentication.
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: This doesn't cover why you'd use WebListener vs IIS. WebListener is for use in services and other self-hosted scenarios that require Windows Authentication. (IIS also supports windows authentication, but you can't self-host it).

File: aspnet/fundamentals/servers.rst [hunk 10]
```diff
@@ -173,88 +169,63 @@ You can add support for WebListener to your ASP.NET application by adding the "M
...
 
 Since Kestrel is open source, it makes an excellent starting point if you need to implement your own custom server. In fact, like all of ASP.NET 5, you're welcome to `contribute <https://github.com/aspnet/KestrelHttpServer/blob/dev/CONTRIBUTING.md>`_ any improvements you make back to the project.
 
-Kestrel currently supports a limited number of feature interfaces, including ``IHttpRequestFeature``, ``IHttpResponseFeature``, and ``IHttpUpgradeFeature``, but additional features will be added in the future. You can see how these interfaces are implemented and supported by Kestrel in its `ServerRequest class <https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Kestrel/ServerRequest.cs>`_, a portion of which is shown below.
+Kestrel currently supports a limited number of feature interfaces, but additional features will be added in the future. You can see how these interfaces are implemented and supported by Kestrel in its ``Frame`` class. For example, the ``IHttpUpgradeFeature`` interface consists of only one property and one method. Kestrel's implementation of this interface is shown here for reference:
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: https://github.com/aspnet/KestrelHttpServer/blob/1.0.0-rc1/src/Microsoft.AspNet.Server.Kestrel/Http/Frame.FeatureCollection.cs#L16

File: aspnet/fundamentals/servers.rst [hunk 11]
```diff
@@ -62,59 +61,46 @@ ASP.NET defines a number of :doc:`request-features` which may be supported on di
...
-	  - Yes
+	  - No*
 	  - No*
 	* - IRequestIdentifierFeature
 	  - Yes
-	  - Yes
 	  - No
 	* - ITlsConnectionFeature
 	  - Yes
 	  - Yes
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: Kestrel got this in RC2

File: aspnet/fundamentals/servers.rst [hunk 12]
```diff
@@ -62,59 +61,46 @@ ASP.NET defines a number of :doc:`request-features` which may be supported on di
...
-	  - No
 	  - Yes (with limits)
 	  - Yes
 	* - IHttpBufferingFeature
 	  - Yes
-	  - Yes
 	  - No
 	* - IHttpConnectionFeature
 	  - Yes
 	  - Yes
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: Kestrel got this in RC2

File: aspnet/fundamentals/servers.rst [hunk 13]
```diff
@@ -173,88 +169,63 @@ You can add support for WebListener to your ASP.NET application by adding the "M
...
 
 Since Kestrel is open source, it makes an excellent starting point if you need to implement your own custom server. In fact, like all of ASP.NET 5, you're welcome to `contribute <https://github.com/aspnet/KestrelHttpServer/blob/dev/CONTRIBUTING.md>`_ any improvements you make back to the project.
 
-Kestrel currently supports a limited number of feature interfaces, including ``IHttpRequestFeature``, ``IHttpResponseFeature``, and ``IHttpUpgradeFeature``, but additional features will be added in the future. You can see how these interfaces are implemented and supported by Kestrel in its `ServerRequest class <https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Kestrel/ServerRequest.cs>`_, a portion of which is shown below.
+Kestrel currently supports a limited number of feature interfaces, but additional features will be added in the future. You can see how these interfaces are implemented and supported by Kestrel in its ``Frame`` class. For example, the ``IHttpUpgradeFeature`` interface consists of only one property and one method. Kestrel's implementation of this interface is shown here for reference:
 
 .. code-block:: c#
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: Remove this, just link to the code.

File: aspnet/fundamentals/servers.rst [hunk 14]
```diff
@@ -173,88 +169,63 @@ You can add support for WebListener to your ASP.NET application by adding the "M
...
+	    await ProduceStartAndFireOnStarting(immediate: true);
+
+	    return DuplexStream;
 	}
 
 Summary
 -------
 
-Because ASP.NET 5 has completely decoupled ASP.NET applications from IIS or any other web server, it's now possible to host ASP.NET applications on any number of different servers. ASP.NET supports three: IIS, WebListener, and Kestrel, which provide two great options for Windows environments and a third, open-source option that can be used on several different operating systems.
+Because ASP.NET 5 has completely decoupled ASP.NET applications from the underlying server, it's now possible to host ASP.NET applications on a number of different servers. ASP.NET supports Kestrel, WebListener, and any custom server that you may implement, which provides great options for Windows, Mac, and Linux.
```
Comment (Tratcher) [inline aspnet/fundamentals/servers.rst]: Behind IIS or IIS Express...

File: aspnet/fundamentals/servers/sample/ServersDemo/src/ServersDemo/project.json [hunk 15]
```diff
@@ -3,15 +3,16 @@
   "version": "1.0.0-*",
 
   "dependencies": {
-    "Microsoft.AspNet.Server.IIS": "1.0.0-beta6",
-    "Microsoft.AspNet.Server.WebListener": "1.0.0-beta6",
-    "Kestrel": "1.0.0-beta6"
+    "Microsoft.AspNet.Hosting.Server.Abstractions": "1.0.0-rc1-final",
```
Comment (Tratcher) [inline aspnet/fundamentals/servers/sample/ServersDemo/src/ServersDemo/project.json]: Abstractions shouldn't be required

File: aspnet/fundamentals/servers.rst [hunk 16]
```diff
@@ -13,42 +13,39 @@ In this article:
...
+ASP.NET 5 does not directly listen for requests, but instead relies on the HTTP server implementation to surface the request to the application as a set of :doc:`feature interfaces <request-features>` composed into an HttpContext. While WebListener is Windows-only; Kestrel is designed to run cross-platform. You can configure your application to be hosted by any or all of these servers by specifying commands in your ``project.json`` file. You can even specify an application entry point for your application, and run it as an executable (using ``dnx run``) rather than hosting it in a separate process.
```
Comment (Rick-Anderson) [inline aspnet/fundamentals/servers.rst]: change Windows-only; to Windows-only, (replace ; with comma)
your ``project.json`` file should be *project.json* file

File: aspnet/fundamentals/servers.rst [hunk 17]
```diff
@@ -13,42 +13,39 @@ In this article:
...
 
-The default web host for ASP.NET applications developed using Visual Studio 2015 is IIS / IIS Express. The "Microsoft.AspNet.Server.IIS" dependency is included in ``project.json`` by default, even with the Empty web site template. Visual Studio provides support for multiple profiles, associated with IIS Express and any other ``commands`` defined in ``project.json``. You can manage these profiles and their settings in the Debug tab of your web application project's Properties menu.
+The default web host for ASP.NET apps developed using Visual Studio 2015 is IIS Express functioning as a reverse proxy server for Kestrel. The "Microsoft.AspNet.Server.Kestrel" and "Microsoft.AspNet.IISPlatformHandler" dependencies are included in ``project.json`` by default, even with the Empty web site template. Visual Studio provides support for multiple profiles, associated with IIS Express and any other ``commands`` defined in ``project.json``. You can manage these profiles and their settings in the Debug tab of your web application project's Properties menu or from the ``launchSettings.json`` file.
```
Comment (Rick-Anderson) [inline aspnet/fundamentals/servers.rst]: ``project.json``.  itallic

Comment (Rick-Anderson) [inline aspnet/fundamentals/servers.rst]: ``launchSettings.json`` file.  ditto

File: aspnet/fundamentals/servers.rst (hunk 16 above)
Comment (Rick-Anderson) [inline aspnet/fundamentals/servers.rst]: Debug tab of your web application project’s Properties menu
Debug and Properties bold

File: aspnet/fundamentals/servers.rst [hunk 18]
```diff
@@ -56,111 +53,108 @@ The ``run`` command will execute the application via its ``void main()`` method
...
+	:emphasize-lines: 3,6-7,10,15
 	:language: c#
 	:dedent: 8
 
+
 IIS and IIS Express
 -------------------
 
-Working with IIS as your server for your ASP.NET application is the default option, and should be familiar to ASP.NET developers who have worked with previous versions of the framework. IIS currently provides support for the largest number of features, and includes IIS management functionality and access to other IIS modules. Hosting ASP.NET 5 on IIS bypasses the legacy ``System.Web`` infrastructure used by prior versions of ASP.NET, providing a substantial performance gain. Most production ASP.NET web applications being deployed to Windows servers will run on IIS.
+IIS is the most feature rich server, and includes IIS management functionality and access to other IIS modules. Hosting ASP.NET 5 on IIS bypasses the legacy ``System.Web`` infrastructure used by prior versions of ASP.NET, providing a substantial performance gain.
```
Comment (Rick-Anderson) [inline aspnet/fundamentals/servers.rst]: Per Elion - remove entire sentence
Hosting ASP.NET 5 on IIS bypasses the legacy ``System.Web`` infrastructure used by prior versions of ASP.NET, providing a substantial performance gain.

File: aspnet/fundamentals/servers.rst [hunk 19]
```diff
@@ -56,111 +53,108 @@ The ``run`` command will execute the application via its ``void main()`` method
...
+
+In ASP.NET 5 on Windows, the web application is hosted by an external process outside of IIS. The HTTP Platform Handler is an IIS 7.5+ module which is responsible for process management of HTTP listeners and used to proxy requests to the processes that it manages.
 
 WebListener
 -----------
 
-WebListener is a Windows-only server that allows ASP.NET applications to be hosted outside of IIS. It runs directly on the `Http.Sys kernel driver <http://www.iis.net/learn/get-started/introduction-to-iis/introduction-to-iis-architecture>`_, and has very little overhead. It supports the same feature interfaces as IIS; in fact, you can think of WebListener as a library version of IIS.
+WebListener is a Windows-only HTTP server for ASP.NET 5. It runs directly on the `Http.Sys kernel driver <http://www.iis.net/learn/get-started/introduction-to-iis/introduction-to-iis-architecture>`_, and has very little overhead.
 
 You can add support for WebListener to your ASP.NET application by adding the "Microsoft.AspNet.Server.WebListener" dependency in project.json and the following command:
```
Comment (Rick-Anderson) [inline aspnet/fundamentals/servers.rst]: *project.json*
italic

**Comment by Rick-Anderson** [comment]
@shirhatti  very close. A couple minor changes and I'll leave the tree rat.
:watch:

**Comment by shirhatti** [comment]
@Rick-Anderson @Tratcher Does it look good to go?

**Comment by Rick-Anderson** [comment]
:shipit:

**Comment by shirhatti** [comment]
@Tratcher Can I merge this?

**Comment by Tratcher** [comment]
:shipit:

**Comment by shirhatti** [comment]
Merged in d1af91392434232cf8e7a4fafceec3895898220d

**Pull request closed without merging**